### ISSUE TITLE
🐛 [BUG] - SubNav: the "Anchor Nav Default Link" variant doesn't provide an `aria-label` on mobile

### ISSUE THREAD

**Issue by victoriaNine** [primer/brand#1023]
### Describe the bug

As reported through these issues:
- https://github.com/github/accessibility-audits/issues/11160
- https://github.com/github/accessibility-audits/issues/11131

The `SubNav` component's ["Anchor Nav Default Link" variant](https://stunning-chainsaw-j82glqz.pages.github.io/?path=/story/components-subnav-features--anchor-nav-default-link-variant) currently doesn't provide a way of setting an `aria-label` value to use for its expand `button`, which currently expects the `aria-current` attribute to be set on one of its link items (see attached video).

As a result, all `SubNav` components currently using that variant trigger this accessibility issue:

![Image](https://github.com/user-attachments/assets/0c8c0e88-c9b2-4f6f-93d3-9e13a2de6bcf)

This causes screen readers to be unable to read out the expand `button`.

----

https://github.com/user-attachments/assets/fdbe5b14-bedb-41d1-bc1b-76f959815e96

### Reproduction steps

```bash
1. Go to https://stunning-chainsaw-j82glqz.pages.github.io/?path=/story/components-subnav-features--anchor-nav-default-link-variant
2. Open the devtools in the 'Elements' tab
3. Select the `SubNav` component's toggle `button`
4. Observe that it doesn't have any readable text
```

### Expected behavior

```bash
1. Go to https://stunning-chainsaw-j82glqz.pages.github.io/?path=/story/components-subnav-features--anchor-nav-default-link-variant
2. Open the devtools in the 'Elements' tab
3. Select the `SubNav` component's toggle `button`
4. The button should contain `Heading` either as its `aria-label` value or as screen reader-only content
```

### Browsers

Mac OS 15.4.1, VoiceOver, Chrome 136.

The initial issues were observed in the following environments:

> OS: Android 14
> Device: Samsung S22
> Application: Google Chrome (Version: 135.0.7049.100)
> Assistive Technology: Talkback

> OS: iOS 18.4.1
> Device: iPhone 13
> Browser: Safari
> Assistive Technology: VoiceOver
